### PR TITLE
10 Prisma schema + initial migration (based on previously created ERD)

### PR DESCRIPTION
**Pull request by marcin98b**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  status          Int           @default(0)
+  user            user          @relation(fields: [user_id], references: [user_id], onDelete: Cascade, map: "user")
+  transaction     transaction[]
+}
+
+model stock {
+  stock_id          Int                 @id
+  company_id        Int
+  quantity          Int
+  price             Decimal             @db.Decimal(16, 2)
```
Comment (Gabryjiel) [inline prisma/schema.prisma]: Price should be integer. Currency should be stored in db in cents or parts of cents.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model buyOffer {
+  buyOffer_id    Int           @id
```
Comment (Gabryjiel) [inline prisma/schema.prisma]: Pick one case: camelCase or pascal_case. Camel case is preferred in JavaScript world.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+model transaction {
+  transaction_id                   Int       @id
+  buyOffer_id                      Int
+  sellOffer_id                     Int
+  status                           Int       @default(0)
+  date                             DateTime  @db.Timestamptz(6)
+  seller_id                        Int
+  buyer_id                         Int
+  buyOffer                         buyOffer  @relation(fields: [buyOffer_id], references: [buyOffer_id], onDelete: Cascade, map: "buyOffer")
+  user_transaction_buyer_idTouser  user      @relation("transaction_buyer_idTouser", fields: [buyer_id], references: [user_id], onDelete: Cascade, map: "buyer")
```
Comment (Gabryjiel) [inline prisma/schema.prisma]: Can these long names be shortened?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,95 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model buyOffer {
```
Comment (Gabryjiel) [inline prisma/schema.prisma]: Model names will look better capitalized.

**Review by Gabryjiel — CHANGES_REQUESTED** [review]

**Comment by Gabryjiel** [comment]
Change target branch into 'develop' not 'main'.

Feature branches should be merged into develop then develop is merged into main.

File: prisma/schema.prisma (hunk 2 above)
Comment (marcin98b) [reply prisma/schema.prisma]: Fixed naming convention in latest commit

File: prisma/schema.prisma (hunk 3 above)
Comment (marcin98b) [reply prisma/schema.prisma]: Fixed relationship naming convention in latest commit

File: prisma/schema.prisma (hunk 1 above)
Comment (marcin98b) [reply prisma/schema.prisma]: Splitted price field into two as suggested
https://github.com/hyter99/AI_22-23_L1/commit/765902511bc6c7687944cc32af4aaa97f86bacb2

Comment (Gabryjiel) [reply prisma/schema.prisma]: No no, I suggested single Integer field.

Instead of 1,50 dollar I would say it's better to store 150 cents.

Comment (marcin98b) [reply prisma/schema.prisma]: sure, np
https://github.com/hyter99/AI_22-23_L1/pull/25/commits/267b7b85310518f938d53990b50f23ddeb7acbd2

**Review by BartoszGrz — APPROVED** [review]

**Pull request closed without merging**